ddtrace/span.py

```diff
@@ -297,8 +300,11 @@ def set_tag(self, key, value=None):
             log.warning("error setting tag %s, ignoring it", key, exc_info=True)
 
     def _set_str_tag(self, key, value):
-        # type: (str, str) -> None
-        self.meta[key] = stringify(value)
+        # type: (Text, AnyStr) -> None
+        try:
+            self.meta[key] = ensure_text(value, errors="replace")
+        except TypeError:
```

Comment:
You should never get a `TypeError` there right?

Reply:
Added a test for the case where this is called with something other than `AnyStr`, e.g. `dict(a=1)`. But I admit this might be more defensive than we ought to be in an internal function.